### ISSUE TITLE
Remove hard-coded account details from TM_Design

### ISSUE THREAD

**Issue by DinisCruz** [TeamMentor/TM_4_0_Design#368]
Currently they are here https://github.com/TeamMentor/TM_4_0_Design/blob/master/node/controllers/Login-Controller.coffee#L4-L6

Note that this will have an impact on a number of UnitTests, which currently use these accounts

This is related to https://github.com/TeamMentor/TM_4_0_Design/issues/366 and ideally should be done in conjunction with https://github.com/TeamMentor/TM_4_0_Design/issues/367 

----
[TM-4.0-Security](https://github.com/TeamMentor/TM_4_0_Design/wiki/TM-4.0-Security)

**Comment by roman87** [comment]
Please remove this asap. Unit tests need to crate their accounts not use hard coded ones. This goes against any best practice we preach. Note that #366 and #367 should not be a per-requisite to this issue.  There should be a simple unit test that creates an account and all other unit tests are to use that created account. I also don't understand why this needs to take 2 days?

**Comment by DinisCruz** [comment]
so my interpretation of **Pint 2d** effort is something that needs a bit of research and thinking, but should ultimately easy to do

the **1d** stuff are things that I can fully visualise the work and can just get on with it (with the actually coding being an 10m to 1h thing)

**Comment by roman87** [comment]
The effort estimate is in time units like days :). It doesn't matter from the estimate perspective if it will take you 1.5 daysto research and 4 hours to code or full 2 days to code.   1 Day stuff is anything that will take you from 10 minutes to a whole day. 2d is anything that will take you from a whole day to two whole days. 
So my question is why removing this will take 2 whole days :)? Ether I don't understand the issue (which is entirely possible) or there is more to the issue than what is written here.

**Comment by DinisCruz** [comment]
Then we should add another category for smaller tasks.

Part of the reason why removing this is not a simple tasks is because a large number of tests rely on the fact that there is a really quick way to login, and once those hard-coded users are removed, it means that every new batch of tests will need to start with a new user being created (which could be speeded up by caching the temp user details somewhere on the .tmCache, but that introduces other issues)

Basically there are a number of moving parts on this one, which in fact will require a couple more Issues to be opened (in order to have a correct representation of the work done).

I'm also being conservative here, since it would expect this to be easy, BUT as you can see, there are quite a lot of moving parts (For example the issue https://github.com/TeamMentor/TM_4_0_Design/issues/348 is much simpler than this one)

**Comment by roman87** [comment]
So what happens if the first test that runs creates a user. And then all other tests use that user? Anyways. It takes as long as it takes, I just want to make sure we do not make #366 and #367 a dependency for this.

**Comment by roman87** [comment]
Bumping this!

**Comment by DinisCruz** [comment]
done with latest commit, hard-coded credentials now only exist on the TM_Design tests (not in the main code base). 

The TM QA tests will need to be updated, but the core changes to TM_Design and its tests are now done

**Comment by roman87** [comment]
looks good. closing
